File: src/exercises/chapter1/lesson14.rb
# frozen_string_literal: true

# S(n) = x + x^3 + x^5 + … + x^2n + 1
# @param {Numeric} x - The base value (can be a real number or integer)
# @param {Integer} n - The number of terms in the series (positive integer)
# @return {Numeric} - The sum of the series x + x^3 + x^5 + … + x^(2n + 1)

def solution(x, n)
  result = 0
  (0..n).each do |i|
    result += x**(2 * i + 1)
  end
  result
end

File: src/exercises/chapter1/Lesson14.ts
/**
 * Lesson 14: Calculate S(n) = x + x^3 + x^5 + … + x^(2n + 1)
 */
export default class Lesson14 {

  /**
   * Calculates the sum S(n) = x + x^3 + x^5 + ... + x^(2n + 1)
   *
   * @param {number} x - The base number
   * @param {number} n - The number of terms in the series
   * @return {number} - The sum of the series
   */
  solution(x: number, n: number): number {
    let result = 0 // Initialize result to accumulate the sum

    // Loop from 0 to n to calculate each term x^(2*i + 1) and add it to the result
    for (let i = 0; i <= n; i++) {
      result += Math.pow(x, 2 * i + 1) // Add x^(2*i + 1) to the cumulative sum
    }

    return result // Return the computed sum
  }
}
